mergify_cli/ci/cli.py

```diff
@@ -0,0 +1,131 @@
+import os
+import re
+import typing
+
+import click
+
+from mergify_cli import console
+from mergify_cli import utils
+from mergify_cli.ci import junit_upload as junit_upload_mod
+
+
+ci = click.Group(
+    "ci",
+    help="Mergify's CI related commands",
+)
+
+
+def get_ci_provider() -> typing.Literal["github_action", "circleci"] | None:
+    if os.getenv("GITHUB_ACTIONS") == "true":
+        return "github_action"
+    if os.getenv("CIRCLECI") == "true":
+        return "circleci"
+    return None
+
+
+def get_job_name() -> str | None:
+    if os.getenv("GITHUB_ACTIONS") == "true":
+        return os.getenv("GITHUB_WORKFLOW")
+    if os.getenv("CIRCLECI") == "true":
+        return os.getenv("CIRCLE_JOB")
+
+    console.log("Error: failed to get the job's name from env", style="red")
+    return None
+
+
+def get_head_sha() -> str | None:
+    if os.getenv("GITHUB_ACTIONS") == "true":
```

Comment:
You do this a lot: if you have to change the way you identify GHA in the future you'll have to change tens of lines.
You should use `get_ci_provider()` instead here.